lib/internal/util.js

```diff
@@ -187,3 +187,18 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+exports.getConstructorOf = function getConstructorOf(obj) {
+  while (obj) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, 'constructor');
```

Comment:
@TimothyGu the code currently explicitly skips over any `get constructor() {}` - previously in `url`, it only used `hasOwnProperty`, so as-is, this is a breaking change for URL. it's only in lib/util that this is keeping the behavior the same.

In other words, any deduplication of this code seems to be a breaking change in one place or the other.